openpathsampling/ensemble.py

```diff
@@ -1636,12 +1641,15 @@ def _find_subtraj_final(self, traj, subtraj_first, ens_num,
         # logger.debug("Can-app " + str(ens.can_append(subtraj, trusted=True)))
         # logger.debug("Call    " + str(ens(subtraj, trusted=True)))
         # TODO: the weird while condition is handling the OVERSHOOTING
```

Comment:
I don't know if this TODO has now to be moved to the if_statemt

Reply:
The TODO is about the fact that the while loop checks both the `can_append` and the `__call__` of the ensemble. This is the issue that some fundamental ensembles (`LengthEnsemble`) know whether you can append to the trajectory without running an additional frame, while others (`AllInXEnsemble`) only know you've left the ensemble after the next frame is generated.